src/WordPress/SubdomainChecker.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace CF\WordPress;
+
+// WordPressAPI::getDomainList could be use in hooks which doesn't 
+// rely on "GET /zones" request SubdomainChecker solves this problem
+// by changing WordPressAPI::domainName
+class SubdomainChecker
+{
+    private $originalDomain = '';
+    private $subDomain = '';
+
+    public function changeDomainName()
+    {
+        $config = new \CF\Integration\DefaultConfig('[]');
+        $logger = new \CF\Integration\DefaultLogger($config->getValue('debug'));
+        $dataStore = new \CF\WordPress\DataStore($logger);
+        $wordpressAPI = new \CF\WordPress\WordPressAPI($dataStore);
+        $wordpressIntegration = new \CF\Integration\DefaultIntegration($config, $wordpressAPI, $dataStore, $logger);
+        $clientAPIClient = new \CF\WordPress\WordPressClientAPI($wordpressIntegration);
+
+        $wpDomain = $wordpressAPI->getOriginalDomain();
+
+        if ($clientAPIClient->isSubdomain($wpDomain)) {
```

Comment:
clientAPIClient does have [isSubdomain](https://github.com/cloudflare/CloudFlare-Wordpress/blob/thellimist/PI-718/src/WordPress/WordPressClientAPI.php#L57-L73). The reason I put there was to avoid another set of objects to be created inside `WordPressAPI` cause I am using `GET /zones` [client](https://github.com/cloudflare/CloudFlare-Wordpress/blob/thellimist/PI-718/src/WordPress/WordPressClientAPI.php#L62) call. 

If you still think it belongs inside `WordPressAPI` I can move it.